src/function/scalar/map/map_keys.cpp

```diff
@@ -0,0 +1,78 @@
+#include "duckdb/planner/expression/bound_function_expression.hpp"
+#include "duckdb/common/string_util.hpp"
+#include "duckdb/parser/expression/bound_expression.hpp"
+#include "duckdb/function/scalar/nested_functions.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/pair.hpp"
+
+namespace duckdb {
+
+static void MapKeysFunction(DataChunk &args, ExpressionState &state, Vector &result) {
+	D_ASSERT(result.GetType().id() == LogicalTypeId::LIST);
+	idx_t count = args.size();
+
+	// Get the arguments vector
+	auto &map = args.data[0];
+	auto &map_keys = MapVector::GetKeys(map);
```

Comment:
From my understanding, this is the only significant difference between the `MapKeysFunction` and the `MapValuesFunction`. I think you can use e.g. a`Functor` for this instead, then we don't have so much duplicate code. You can have a look at `list_aggregates.cpp` for this.